anitya/lib/backends/maven.py

```diff
@@ -70,3 +75,96 @@ def get_versions(cls, project):
                           .replace(':', '/'))
 
         return get_versions_by_regex(url, REGEX, project)
+
+    @classmethod
+    def verify_url(cls, group_id, artifact_id):
+        match_decimal = '(?=\d)'
+        match_non_decimal = r'(?=\D)'
+
+        # match name like like artifact_id23.23_1
+        temp_artifact = do_replace(artifact_id, match_decimal, match_decimal)
```

Comment:
`r'(?=\d)' and `r'(?=\D)'` are both non capturing regexps. They should match decimal/non decimal characters with capturing them. What I am trying to achieve is replace `dots` for `slashes` in `artifactId` and `groupId`. In theory `replace('.', '/')` should be enough. However it's harder, because URL can contain numbers with `dots` [1],
Another thing is that some URL's don't have slashes instead of dots [2].
So I tried to create approach which tests all combinations.

[1] http://repo2.maven.org/maven2/com/zenecture/neuroflow-application_2.12/
[2] http://repo2.maven.org/maven2/com/liferay/com.liferay.amazon.rankings.web/